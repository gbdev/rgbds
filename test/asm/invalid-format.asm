println STRFMT("%+d %++d", 42, 42)
println STRFMT("%#x %##x", 42, 42)
println STRFMT("%-4d %--4d", 42, 42)
println STRFMT("%.f %..f", 42.0, 42.0)

DEF N = 42
println "{5d:N} {5d5:N}"
println "{x:N} {xx:N}"

println STRFMT("%+s", "hello")
println STRFMT("%#s", "hello")
println STRFMT("%0s", "hello")
println STRFMT("%.5s", "hello")

println STRFMT("%#d", 42)
println STRFMT("%.5d", 42)
