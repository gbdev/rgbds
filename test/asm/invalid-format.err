error: invalid-format.asm(1):
    STRFMT: Invalid format spec for argument 2
error: invalid-format.asm(2):
    STRFMT: Invalid format spec for argument 2
error: invalid-format.asm(3):
    STRFMT: Invalid format spec for argument 2
error: invalid-format.asm(4):
    STRFMT: Invalid format spec for argument 2
error: invalid-format.asm(7):
    Invalid format spec '5d5'
error: invalid-format.asm(8):
    Invalid format spec 'xx'
error: invalid-format.asm(10):
    Formatting string with sign flag '+'
error: invalid-format.asm(11):
    Formatting string with prefix flag '#'
error: invalid-format.asm(12):
    Formatting string with padding flag '0'
error: invalid-format.asm(13):
    Formatting string with fractional width
error: invalid-format.asm(15):
    Formatting type 'd' with prefix flag '#'
error: invalid-format.asm(16):
    Formatting type 'd' with fractional width
error: Assembly aborted (12 errors)!
