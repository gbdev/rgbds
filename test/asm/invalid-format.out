+42 %d
$2a %x
42   %4d
42 %f
   42    42
2a 2a
hello
hello
hello
hello
42
42
